osu.Game/Skinning/GameplaySkinComponentLookup.cs

```diff
@@ -25,7 +27,41 @@ public GameplaySkinComponentLookup(T component)
             Component = component;
         }
 
-        protected virtual string RulesetPrefix => string.Empty;
-        protected virtual string ComponentName => Component.ToString();
+        public virtual RulesetInfo? Ruleset => null;
+
+        public override string ToString()
+        {
+            if (Ruleset == null)
+                return Component.ToString();
+
+            return $"{Component.ToString()} (\"{Ruleset.Name}\" only)";
+        }
+
+        public bool Equals(GameplaySkinComponentLookup<T>? other)
+        {
+            if (ReferenceEquals(null, other)) return false;
+            if (ReferenceEquals(this, other)) return true;
+
+            return EqualityComparer<T>.Default.Equals(Component, other.Component);
+        }
+
+        public override bool Equals(object? obj)
+        {
+            if (ReferenceEquals(null, obj)) return false;
+            if (ReferenceEquals(this, obj)) return true;
+            if (obj.GetType() != GetType()) return false;
+
+            return Equals((GameplaySkinComponentLookup<T>)obj);
+        }
+
+        public override int GetHashCode()
+        {
+            return EqualityComparer<T>.Default.GetHashCode(Component);
+        }
```

Comment:
Are these required?

Reply:
The `object` ones specifically? Doesn't look to be - will remove. The others are used in `SkinEditor` - at least the interface one is.

Reply:
Yep, meant the `object`-level comparers/hashcodes.

Reply:
Actually, it looks like these are required, because the `SkinEditor` dropdown dedupes on these. Haven't observed `Equals(object)` being used, but I have observed `GetHashCode()`.

Have restored both for good practice.